app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -261,3 +306,181 @@ export async function circularRedirect(query: string) {
     throw redirect(circularURL)
   }
 }
+
+/**
+ * Gets all entries in circulars_history for a given circularId
+ * @param circularId
+ * @returns an array of previous versions of a Circular sorted by version
+ */
+export async function getVersions(circularId: number): Promise<number[]> {
+  const db = await tables()
+  const versions = (
+    await db.circulars_history.query({
+      KeyConditionExpression: 'circularId = :circularId',
+      ExpressionAttributeValues: {
+        ':circularId': circularId,
+      },
+      ProjectionExpression: 'version',
+    })
+  ).Items
+
+  return [...versions.map((x) => x.version), versions.length + 1]
```

Comment:
Note that this will return `[1]` if (a) the Circular exists but has only one version or (b) the Circular does not exist. Do we care to be correct for the latter corner case?

Reply:
I think the latter shouldn't be that much of an issue, since it is just returning an array, it won't really mean anything until we put it into further context, like generating links to view the versions